azuread/table_azuread_application.go

```diff
@@ -35,145 +43,214 @@ func tableAzureAdApplication() *plugin.Table {
+			{Name: "tags_src", Type: proto.ColumnType_JSON, Description: "Custom strings that can be used to categorize and identify the application.", Transform: transform.FromMethod("GetTags")},
+			{Name: "web", Type: proto.ColumnType_JSON, Description: "Specifies settings for a web application.", Transform: transform.FromMethod("ApplicationWeb")},
+
+			// Standard columns
+			{Name: "tags", Type: proto.ColumnType_JSON, Description: ColumnDescriptionTags, Transform: transform.From(adApplicationTags)},
+			{Name: "title", Type: proto.ColumnType_STRING, Description: ColumnDescriptionTitle, Transform: transform.From(adApplicationTitle)},
+			{Name: "tenant_id", Type: proto.ColumnType_STRING, Description: ColumnDescriptionTenant, Hydrate: plugin.HydrateFunc(getTenant).WithCache(), Transform: transform.FromValue()},
 		},
 	}
 }
 
 //// LIST FUNCTION
 
 func listAdApplications(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
-	session, err := GetNewSession(ctx, d)
+	// Create client
+	client, adapter, err := GetGraphClient(ctx, d)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error creating client: %v", err)
 	}
 
-	client := msgraph.NewApplicationsClient(session.TenantID)
-	client.BaseClient.Authorizer = session.Authorizer
-
-	input := odata.Query{}
+	// List operations
+	input := &applications.ApplicationsRequestBuilderGetQueryParameters{}
 
 	// Restrict the limit value to be passed in the query parameter which is not between 1 and 999, otherwise API will throw an error as follow
 	// unexpected status 400 with OData error: Request_UnsupportedQuery: Invalid page size specified: '1000'. Must be between 1 and 999 inclusive.
 	limit := d.QueryContext.Limit
 	if limit != nil {
 		if *limit > 0 && *limit <= 999 {
-			input.Top = int(*limit)
+			l := int32(*limit)
+			input.Top = &l
 		}
 	}
 
-	qualsColumnMap := []QualsColumn{
-		{"app_id", "string", "appId"},
-		{"display_name", "string", "displayName"},
-		{"publisher_domain", "string", "publisherDomain"},
+	var queryFilter string
+	equalQuals := d.KeyColumnQuals
+	filter := buildApplicationQueryFilter(equalQuals)
+
+	if equalQuals["filter"] != nil {
+		queryFilter = equalQuals["filter"].GetStringValue()
 	}
 
-	filter := buildCommaonQueryFilter(qualsColumnMap, d.Quals)
-	if len(filter) > 0 {
-		input.Filter = strings.Join(filter, " and ")
+	if queryFilter != "" {
+		input.Filter = &queryFilter
+	} else if len(filter) > 0 {
+		joinStr := strings.Join(filter, " and ")
+		input.Filter = &joinStr
 	}
 
-	applications, _, err := client.List(ctx, input)
+	options := &applications.ApplicationsRequestBuilderGetRequestConfiguration{
+		QueryParameters: input,
+	}
+
+	result, err := client.Applications().GetWithRequestConfigurationAndResponseHandler(options, nil)
 	if err != nil {
-		if isNotFoundError(err) {
-			return nil, nil
-		}
-		return nil, err
+		errObj := getErrorObject(err)
+		return nil, errObj
```

Comment:
Can you please add an error logging message here for better logging support (and to all other places we return an error)?